server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -466,4 +496,31 @@ long count() {
     int numRegisteredCloseListeners() { // for testing
         return registeredClosedListeners.size();
     }
+
+    /**
+     * Creates a new disk tier instance. Should only be run if the instance will actually be used!
+     * Otherwise, it may not be closed properly.
+     * @return A new disk tier instance
+     */
+    @Override
+    public EhCacheDiskCachingTier<Key, BytesReference> getDiskTier() {
```

Comment:
getDiskTier method creating a new instance everytime could be misleading and I dont think we should rely on comments only to warn of its usage, its very easy to just call this without reading the code comments. 
How about we rename this to something that is self explanatory ? Like getNewDiskTier or something similar ?